gui/webdisplay/src/TWebWindow.cxx

```diff
@@ -482,26 +447,34 @@ void ROOT::Experimental::TWebWindow::Send(const std::string &data, unsigned conn
    CheckDataToSend();
 }
 
+///////////////////////////////////////////////////////////////////////////////////
+/// Sends data to specified connection
+/// If connid==0, data will be send to all connections
+
+void ROOT::Experimental::TWebWindow::Send(unsigned connid, const std::string &data)
+{
+   SubmitData(connid, true, std::string(data), 1);
+}
+
 ///////////////////////////////////////////////////////////////////////////////////
 /// Send binary data to specified connection
 /// If connid==0, data will be sent to all connections
 
-void ROOT::Experimental::TWebWindow::SendBinary(std::shared_ptr<RawBuffer> &data, unsigned connid, unsigned chid)
+void ROOT::Experimental::TWebWindow::SendBinary(unsigned connid, std::string &&data)
 {
-   for (auto &&conn : fConn) {
-      if (connid && connid != conn.fConnId)
-         continue;
+   SubmitData(connid, false, std::move(data), 1);
+}
 
-      if (conn.fQueue.empty() && (conn.fSendCredits > 0)) {
-         SendBinaryDataViaConnection(conn, chid, data);
-      } else if (conn.fQueue.size() < fMaxQueueLength) {
-         conn.fQueue.emplace_back(chid, data);
-      } else {
-         R__ERROR_HERE("webgui") << "Maximum queue length achieved";
-      }
-   }
+///////////////////////////////////////////////////////////////////////////////////
+/// Send binary data to specified connection
+/// If connid==0, data will be sent to all connections
 
-   CheckDataToSend();
+void ROOT::Experimental::TWebWindow::SendBinary(unsigned connid, const void *data, std::size_t len)
+{
+   std::string buf;
+   buf.resize(len);
+   memcpy((void *)buf.data(), data, len);
```

Comment:
This version is bounds-safer and should be as performant:
```
buf.reserve(len);
std::copy(data, data + len, buf.begin());
```

Reply:
Well as I said before: if you have an array of bytes then it's `const char*`. If you have a pointer to an unknown const type it's `const void*`. Here it's bytes, and thus this works: 
```
#include <string>
#include <utility>

void foo(const char *data, std::size_t len) {
    std::string buf;
    buf.resize(len);
    std::copy(data, data+len, buf.begin());
}
```

Reply:
Finally - https://github.com/linev/root/commit/e792b7ee8691ddc59af2b4c4e5a51290ce48f074